src/features/device-provisioning/register.ts

```diff
@@ -43,6 +42,8 @@ export const register: RequestHandler = async (req, res) => {
 			throw new BadRequestError('UUID must be specified');
 		}
 
+		await gracefullyDenyConflictingRegistrations(uuid, req.body.api_key);
```

Comment:
If the following calls fail the cache has been written but the uuid is potentially not registered? Can it be possible that the rules fail the TX but the cache has already save the uuid / api_key and will report it conflicting?
Does the cache needs a cleanup in the catch?